docs/reference/health/health.asciidoc

```diff
@@ -24,42 +28,217 @@ NOTE: {cloud-only}
 [[health-api-desc]]
 ==== {api-description-title}
 
-The health API returns a the health status of an Elasticsearch cluster. It
+The health API returns the health status of an Elasticsearch cluster. It
 returns a list of components that compose Elasticsearch functionality. Each
 component's health is determined by health indicators associated with the
 component.
```

Comment:
Shall we be a bit more verbose here and mention impacts and user actions? (briefly describing all the basic concepts of the API)